### ISSUE TITLE
Unable to parse PDF from an existing PDF on web

### ISSUE THREAD

**Issue by garrettApproachableGeek** [DavBfr/dart_pdf#1065]
**Describe the bug**

When using the following code on web, we are unable to load particular pdf documents
Using: 
`pdf_crypto: ^3.2.6`

**To Reproduce**
Code snippet to reproduce the behavior:
```
pdf_crypto.PdfDocumentParser(bytes),
```

**Expected behavior**

The PDF document is parsed and returns a `PdfDocumentParserBase`

**Flutter Doctor**

```
[✓] Flutter (Channel stable, 2.10.2, on macOS 12.2.1 21D62 darwin-arm, locale en-US)
    • Flutter version 2.10.2 at /Users/garrett/fvm/versions/2.10.2
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 097d3313d8 (3 months ago), 2022-02-18 19:33:08 -0600
    • Engine revision a83ed0e5e3
    • Dart version 2.16.1
    • DevTools version 2.9.2

[✓] Android toolchain - develop for Android devices (Android SDK version 32.1.0-rc1)
    • Android SDK at /Users/garrett/Library/Android/sdk
    • Platform android-32, build-tools 32.1.0-rc1
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7772763)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.3.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7772763)

[✓] VS Code (version 1.67.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.40.0

[✓] Connected device (3 available)
    • Garrett’s iPad (mobile) • 00000000-0000000000000000 • ios            • iOS 15.3.1 19D52
    • macOS (desktop)         • macos                     • darwin-arm64   • macOS 12.2.1 21D62 darwin-arm
    • Chrome (web)            • chrome                    • web-javascript • Google Chrome 102.0.5005.61

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```

**Desktop (please complete the following information):**
 - [ ] iOS
 - [ ] Android
 - [X] Browser
 - [ ] Windows
 - [ ] Linux

**Chrome:**
 - Device: [Macbook Pro M1 Pro]
 - OS: [MacOS]
 - Browser [Chrome]
 - Version [Version 102.0.5005.61 (Official Build) (arm64)]

**Additional context**

[example.pdf](https://github.com/DavBfr/dart_pdf/files/8788012/example.pdf)

This is where I think the stack trace leads to
```
var decoded = Uint8List.fromList(ZLibDecoder().convert(data));
```
StackTrace
```
Uncaught (in promise) Error: Unsupported operation: _newZLibInflateFilter
    at Object.throw_ [as throw] (errors.dart:251:49)
    at RawZLibFilter._makeZLibInflateFilter (io_patch.dart:669:5)
    at new io._ZLibDecoderSink.__ (data_transformer.dart:509:27)
    at io.ZLibDecoder.new.startChunkedConversion (data_transformer.dart:409:16)
    at io.ZLibDecoder.new.convert (data_transformer.dart:394:5)
    at [_decodeFilter] (parsed_object_stream.dart:47:56)
    at parsed_object_stream.PdfParsedObjectStream.new.decode (parsed_object_stream.dart:98:14)
    at Object.compressedCrossRefParser (cross_ref_parser.dart:169:21)
    at Object.crossRefParser (cross_ref_parser.dart:99:15)
    at new document$.PdfDocumentParser.new (document.dart:61:12)
    at pdf_to_image.PdfToImage.new.parsePdfToImage (pdf_to_image.dart:39:20)
    at parsePdfToImage.next (<anonymous>)
    at runBody (async_patch.dart:84:54)
    at Object._async [as async] (async_patch.dart:123:5)
    at pdf_to_image.PdfToImage.new.parsePdfToImage (pdf_to_image.dart:19:39)
    at pdf_drawing_dialog._PdfDrawingDialogState.new.selectPdfFiles (pdf_drawing_dialog.dart:437:48)
    at selectPdfFiles.next (<anonymous>)
    at async_patch.dart:45:50
    at _RootZone.runUnary (zone.dart:1685:54)
    at _FutureListener.thenAwait.handleValue (future_impl.dart:159:18)
    at handleValueCallback (future_impl.dart:766:44)
    at _Future._propagateToListeners (future_impl.dart:795:13)
    at [_completeWithValue] (future_impl.dart:601:5)
    at async._AsyncCallbackEntry.new.callback (future_impl.dart:639:7)
    at Object._microtaskLoop (schedule_microtask.dart:40:11)
    at _startMicrotaskLoop (schedule_microtask.dart:49:5)
    at async_patch.dart:166:15
```

**Comment by garrettApproachableGeek** [comment]
We could swap `var decoded = Uint8List.fromList(ZLibDecoder().convert(data));` to allow this to work for web `var decoded = Uint8List.fromList(ZLibDecoder().decodeBytes(data));` using the `archive` package which I see is also used by some of the other pdf packages. To keep the speed of the decode for iOS/Android we could use `dart:io` or `archive_io` conditionally based on platform

**Comment by DavBfr** [comment]
Yes, that's the idea.

**Comment by DavBfr** [comment]
Fixed in version 3.2.7
